### PR TITLE
fix: Twilo attachment download fallback to the step where authorization headers are not passed.

### PR DESCRIPTION
**Pull request by muhsin-k**
Fixes https://linear.app/chatwoot/issue/CW-2911/downclienterror-400-bad-request-downclienterror

**Where did it impact, how much did it impact?**

In the previous release, we enabled "HTTP Basic Authentication" to ensure that all attachments requiring HTTP authentication are secured. This is particularly important for media files that may contain sensitive data, as recommended by Twilio. However, some users experienced issues because they did not enable this option, despite our alerts prompting them to do so. (There are 2.6K events in sentry).

**Solution**

If the authenticated attachment download call fails, add another call to download the attachment without authentication.

### CODE REVIEW

File: app/services/twilio/incoming_message_service.rb [hunk 1]
```diff
@@ -126,4 +122,26 @@ def attach_files
...
+    Down.download(
+      params[:MediaUrl0],
+      # https://support.twilio.com/hc/en-us/articles/223183748-Protect-Media-Access-with-HTTP-Basic-Authentication-for-Programmable-Messaging
+      http_basic_authentication: [twilio_channel.account_sid, twilio_channel.auth_token || twilio_channel.api_key_sid]
+    )
+  end
+
+  def handle_download_attachment_error(error)
+    if error.message.include?('401 Unauthorized')
+      Down.download(params[:MediaUrl0])
```
Comment (scmmishra) [inline app/services/twilio/incoming_message_service.rb:142]: What if this has an error? That is not handled right?

Comment (scmmishra) [inline app/services/twilio/incoming_message_service.rb:142]: We probably should log something to track the retry

File: app/services/twilio/incoming_message_service.rb [hunk 2]
```diff
@@ -126,4 +122,26 @@ def attach_files
...
+  def download_with_auth
+    Down.download(
+      params[:MediaUrl0],
+      # https://support.twilio.com/hc/en-us/articles/223183748-Protect-Media-Access-with-HTTP-Basic-Authentication-for-Programmable-Messaging
+      http_basic_authentication: [twilio_channel.account_sid, twilio_channel.auth_token || twilio_channel.api_key_sid]
+    )
+  end
+
+  def handle_download_attachment_error(error)
+    if error.message.include?('401 Unauthorized')
```
Comment (scmmishra) [inline app/services/twilio/incoming_message_service.rb:141]: Is there a way we can get an HTTP error code instead of checking against a string?

**Review by scmmishra — CHANGES_REQUESTED** [review]

File: app/services/twilio/incoming_message_service.rb (hunk 1 above)
Comment (muhsin-k) [reply app/services/twilio/incoming_message_service.rb:142]: We are encountering the 401 Unauthorized error for users who have not yet enabled media protection. We will exclude all other codes.
@sojan-official thoughts?

Comment (muhsin-k) [reply app/services/twilio/incoming_message_service.rb:142]: Done

**Pull request merged**